Learn To Fly 3 - Stuck after Clicking on Story Mode

### Describe the bug

Loading SWF file https://www.spiele-umsonst.de/azad/downloads/flashgames/learntofly3.swf
ruffle_web-wasm_extensions.js:681 ERROR core/src/avm2/events.rs:419 Error dispatching event EventObject(EventObject { type: "enterFrame", class: flash.events::Event, ptr: 0x70f0d60 }) to handler FunctionObject(FunctionObject { ptr: 0x5a4bf80 }) : RustError("VerifyError: Trait concat in class ColorMatrix has same name as trait concat in class Array, but does not override it")

Here is the error I get when I want to start the story mode

### Expected behavior

Start it?

### Affected platform

Online demo

### Operating system

All

### Browser

All

### Additional information

_No response_

--- a/core/src/avm2/globals/flash/filters/ColorMatrixFilter.as
+++ b/core/src/avm2/globals/flash/filters/ColorMatrixFilter.as
@@ -16,17 +16,20 @@
 
 		// From the Flash docs, we need to make a copy of the `Array`,
 		// as modifying the `filter.matrix` directly should have no effect.
+		// We call the method in the AS3 namespace, as some SWFS define a
+		// extend `Array` and declare a *public* 'concat' method with a
+		// different signature.
 
 		public function get matrix(): Array {
-			return this._matrix.concat();
+			return this._matrix.AS3::concat();
 		}
 
 		public function set matrix(matrix:Array):void {
-			this._matrix = matrix.concat();
+			this._matrix = matrix.AS3::concat();
 		}
 
 		override public function clone(): BitmapFilter {
-			return new ColorMatrixFilter(this.matrix.concat());
+			return new ColorMatrixFilter(this.matrix.AS3::concat());
 		}
 	}
 }

